src/hypothesis/searchstrategy/regex.py

```diff
@@ -0,0 +1,352 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import re
+import sys
+import sre_parse as sre
+
+import hypothesis.errors as he
+import hypothesis.strategies as hs
```

Comment:
I’m not going to push too hard on this, but elsewhere Hypothesis tends to import the objects rather than sticking a short prefix on like this, e.g.

```python
from hypothesis.errors import InvalidArgument
```

I’d prefer that, but not strongly.